src/main/java/net/dv8tion/jda/core/requests/restaction/ChannelAction.java

```diff
@@ -0,0 +1,178 @@
+/*
+ *     Copyright 2015-2016 Austin Keener & Michael Ritter
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.core.requests.restaction;
+
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.Channel;
+import net.dv8tion.jda.core.entities.EntityBuilder;
+import net.dv8tion.jda.core.entities.Guild;
+import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.entities.Role;
+import net.dv8tion.jda.core.requests.Request;
+import net.dv8tion.jda.core.requests.Response;
+import net.dv8tion.jda.core.requests.RestAction;
+import net.dv8tion.jda.core.requests.Route;
+import org.apache.http.util.Args;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONString;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @since  3.0
+ * @author Florian Spieß
+ */
+public class ChannelAction extends RestAction<Channel> //todo documentation
+{
+    public static final int ROLE_TYPE = 0;
+    public static final int USER_TYPE = 1;
+
+    protected final Set<PromisePermissionOverride> overrides = new HashSet<>();
+    protected final Guild guild;
+    protected final boolean voice;
+    protected String name;
+
+    // --voice only--
+    protected Integer bitrate = null;
+    protected Integer userlimit = null;
+
+    public ChannelAction(JDA api, Route.CompiledRoute route, String name, Guild guild, boolean voice)
```

Comment:
`name` provided is never actually used.

With that said, if we want to keep the `.createTextChannel(String name)` method, then we should have it return `new ChannelAction(...).setName(name);`
I.e: Remove the `name` parameter from ChannelAction's constructor.